### PR TITLE
Updated ShowTypingMiddleware to re throw errors

### LINKED ISSUE #726: ShowTypingMiddleware suppresses errors and does not allow adapter.onTurnError to handle them
## Versions
What package version of the SDK are you using.
4.2.0
What nodejs version are you using
8.11.1
What browser version are you using
N/A
What os are you using
MacOS

## Describe the bug
I have a onTurnError handler and ShowTypingMiddleware attached to the botframework adapter.
```
    adapter = new BotFrameworkAdapter(adapterSettings)
        .use(new ShowTypingMiddleware());

    // Catch-all for errors.
    adapter.onTurnError = onTurnError;
```
I've done some "digging" in the source code and found out this line 
```
return await next().then(stopInterval, stopInterval);
```
which just stops the interval in error/success scenario and does not allows the error to propagate to the adapter which normally "fires" onTurnError handler function.

The fix: 
```
function stopInterval(): void {
    finished = true;
    if (hTimeout) { clearTimeout(hTimeout); }
}

function stopIntervalWithError(err: Error): void {
    stopInterval();
    throw err; //  here the error is re-thrown
}
return await next().then(stopInterval, stopIntervalWithError);
```

## To Reproduce
N/A

## Expected behavior
BotFrameworkAdapter to "fire" the provided onTurnError function.

I would gladly contribute with a pull request, but following the guidelines https://github.com/Microsoft/botbuilder-js/wiki/Contribution-Guidelines

when I try to push the new created branch, I encounter the following

git push -u origin fix/typing-middleware-suppresses-errors
ERROR: Permission to Microsoft/botbuilder-js.git denied to jovanmilenkoski.

[bug]

### PR DESCRIPTION
**Pull request by garypretty**
Fixes #726 

Typing indicator works as normal, unless an error is thrown, in which case it is now re thrown to the adapter.

### CODE REVIEW

**Comment by johnataylor** [comment]
@garypretty we had another fix in mind.  rather than mixing the await and the then() why not wrap the await next call in a try... finally and call the stopInterval from there.

File: libraries/botbuilder-core/src/showTypingMiddleware.ts [hunk 1]
```diff
@@ -98,7 +103,7 @@ export class ShowTypingMiddleware implements Middleware {
 
         // Let the rest of the process run.
         // After everything has run, stop the indicator!
-        return await next().then(stopInterval, stopInterval);
+        return await next().then(stopInterval, stopIntervalWithError);
```
Comment (johnataylor) [inline libraries/botbuilder-core/src/showTypingMiddleware.ts]: discussed with @Stevenic and his suggested fix was more like this:

```ts
        try
        {
              return await next();
         }
        finally
        {
               stopInterval();
        }
```

i.e. similar to the C#

Comment (EricDahlvang) [reply libraries/botbuilder-core/src/showTypingMiddleware.ts]: @johnataylor  Wouldn't a try/finally still swallow all exceptions that occur within 'next()'?  The goal of the fix is to propagate those exceptions up the stack, not catch and swallow them.

Comment (johnataylor) [reply libraries/botbuilder-core/src/showTypingMiddleware.ts]: @ERicDahlvang that's not my understanding. this should behave exactly like the finally in C#. in other words adding a catch will stop the exception but a finally just gets run as the exception is propagated

Comment (EricDahlvang) [reply libraries/botbuilder-core/src/showTypingMiddleware.ts]: Oh, there's no catch.  Sorry, I missed that... carry on. :)

**Pull request merged**